docker-compose.yml

```diff
@@ -15,732 +15,779 @@
 
-  cpp-fedora:
-    # Usage:
-    #   docker-compose build cpp-fedora
-    #   docker-compose run cpp-fedora
-    image: arrowdev/arrow-cpp-fedora:latest
-    shm_size: 2G
+  conda-python:
+    image: ${ORG}/${ARCH}-conda-latest-python-${PYTHON}:latest
     build:
       context: .
-      dockerfile: cpp/Dockerfile.fedora
-    environment:
-      PARQUET_TEST_DATA: /arrow/cpp/submodules/parquet-testing/data
-    volumes: *fedora-volumes
+      dockerfile: ci/docker/conda-latest-python.dockerfile
+      cache_from:
+        - ${ORG}/${ARCH}-conda-latest-python-${PYTHON}:latest
+      args:
+        org: ${ORG}
+        arch: ${ARCH}
+        conda: latest
+        python: ${PYTHON}
+    shm_size: *shm-size
+    volumes: *conda-volumes
+    entrypoint: *conda-entrypoint
+    command: &python-conda-command
+      ["/arrow/ci/scripts/cpp_conda_build.sh /arrow /build/cpp &&
+        /arrow/ci/scripts/python_build.sh /arrow &&
+        /arrow/ci/scripts/python_test.sh /arrow"]
+
+  cuda-python:
+    image: ${ORG}/${ARCH}-cuda-${CUDA}-python-3:latest
+    build:
+      context: .
+      dockerfile: ci/docker/linux-apt-python-3.dockerfile
+      cache_from:
+        - ${ORG}/${ARCH}-cuda-${CUDA}-python-3:latest
+      args:
+        base: ${ORG}/${ARCH}-cuda-${CUDA}-cpp:latest
+    shm_size: *shm-size
+    volumes: *cuda-volumes
+    command: &python-command >
+      /bin/bash -c "
+        /arrow/ci/scripts/cpp_build.sh /arrow /build/cpp &&
+        /arrow/ci/scripts/python_build.sh /arrow &&
+        /arrow/ci/scripts/python_test.sh /arrow"
+
+  debian-python:
+    image: ${ORG}/${ARCH}-debian-${DEBIAN}-python-3:latest
+    build:
+      context: .
+      dockerfile: ci/docker/linux-apt-python-3.dockerfile
+      cache_from:
+        - ${ORG}/${ARCH}-debian-${DEBIAN}-python-3:latest
+      args:
+        base: ${ORG}/${ARCH}-debian-${DEBIAN}-cpp:latest
+    shm_size: *shm-size
+    volumes: *debian-volumes
+    command: *python-command
 
-  cpp-debian-testing:
-    # Usage:
-    #   docker-compose build cpp-debian-testing
-    #   docker-compose run cpp-debian-testing
-    image: arrowdev/arrow-cpp-debian-testing:latest
-    shm_size: 2G
+  ubuntu-python:
+    image: ${ORG}/${ARCH}-ubuntu-${UBUNTU}-python-3:latest
     build:
       context: .
-      dockerfile: cpp/Dockerfile.debian-testing
-    environment:
-      PARQUET_TEST_DATA: /arrow/cpp/submodules/parquet-testing/data
-    volumes: *debian-testing-volumes
+      dockerfile: ci/docker/linux-apt-python-3.dockerfile
+      cache_from:
+        - ${ORG}/${ARCH}-ubuntu-${UBUNTU}-python-3:latest
+      args:
+        base: ${ORG}/${ARCH}-ubuntu-${UBUNTU}-cpp:latest
+    shm_size: *shm-size
+    volumes: *ubuntu-volumes
+    command: *python-command
 
-  cpp-ubuntu-trusty:
-    # Usage:
-    #   docker-compose build cpp-ubuntu-trusty
-    #   docker-compose run cpp-ubuntu-trusty
-    image: arrowdev/arrow-cpp-ubuntu-trusty:latest
-    shm_size: 2G
+  fedora-python:
```

Comment:
Do we need to test all OS for python, I'd say only C++ needs to be tested for different linux distributions.

Reply:
We don't need to, but it doesn't harm to have it around. We can choose later to perhaps have nightly tests.

Reply:
Added to the nightly release candidates list.